### ISSUE TITLE
Check for slow transitions between pages (first load)

### ISSUE THREAD

**Issue by romanchyla** [adsabs/bumblebee#1737]
This might not be totally obvious, but the PR https://github.com/adsabs/bumblebee/pull/1735 made the page display contingent of successful load of all components that make the page (so if one of them failed, the whole page might fail to load)

It is about rendering, so it doesn't depend on api requests

It will harder to detect, but should be checked

**Comment by romanchyla** [comment]
btw: the disappearing of the widget is likely due to some other components not being loaded (so there is a delay) - but only first time, so an alternative solution of pre-fetching might work: `bbb.getWidget('DetailsPage').done(function(w) {w.assemble(bbb)})`

**Comment by thostetler** [comment]
yeah I tried to do the prefetching, but wasn't sure where I should put that line -- where were you thinking this should go?  Also, the rendering was already contingent on everything loading, since the page manager switch doesn't happen until all the promises succeed, I just make the actual dom swap happen at that moment as well, probably there is still a small flash that happens.
